source/Halibut/ServiceModel/DelegateServiceFactory.cs

```diff
@@ -22,6 +23,33 @@ public DelegateServiceFactory Register<TContract>(Func<TContract> implementation
             return this;
         }
 
+        public DelegateServiceFactory Register<TContract, TAsyncContract>(Func<TAsyncContract> implementation)
+        {
+            AsyncServiceVerifier.VerifyAsyncSurfaceAreaFollowsConventions<TContract, TAsyncContract>();
+            
+            var serviceType = typeof(TContract);
+            services.Add(serviceType.Name, () => implementation());
+            lock (serviceTypes)
+            {
+                serviceTypes.Add(serviceType);
+            }
+
+            return this;
+        }
+        
+        // Used for testing badly-registered sync/async contracts
+        internal DelegateServiceFactory RegisterWithNoVerification<TContract, TAsyncContract>(Func<TAsyncContract> implementation)
```

Comment:
Would it make sense to make a special _test_ service factory which allows this?

Reply:
That's a good idea 👍🏻 I'll do that

Reply:
Done. Clunky, but done!